js/slider.js

```diff
@@ -0,0 +1,65 @@
+(function(ts) {
+  CRM.$(function($) {
+    $('select.volunteer_slider').each(function(){
+      $(this).select2('destroy').hide().after(buildSlider($(this)));
+    });
+
+    function buildSlider(sel) {
+      var selID = sel.attr('id');
+      var required = sel.is('.required');
+
+      var container = $('<div>', {
+        "class": "volunteer_slider",
+        "data-original-id": selID
+      });
+
+      container.slider({
+        /**
+         * Update the hidden form elements after the user drops the slider
+         */
+        change: function(event, ui) {
+          // wipe out all previous selections
+          $('#' + selID).find('option').prop('selected', false);
+
+          // select all options up to where the user dropped the slider
+          for (var i = 0; i <= ui.value; i++) {
+            var option = $('#' + selID).find('option')[i];
+            $(option).prop('selected', true);
+          }
+        },
+        min: (required ? 0 : -1),
```

Comment:
Add comment: Set to -1 to allow for deselecting the minimum option when field is not required.